### PR TITLE
[PTCD-508] Add bulk-upload test coverage relevant to upcoming change

### PR DESCRIPTION
**Pull request by timabell**
This adds tests showing how multiple csv rows are combined into individual apprenticeship records with multiple child locations.

It also adds more detailed coverage around delivery modes, which is where a change in behaviour has been requested.

* Move theory attributes to be next to the test method signature to make it clearer that it is describing the method as defining a theory.

### CODE REVIEW

File: Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFact.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using Dfc.CourseDirectory.Core.Models;
+
+namespace Dfc.CourseDirectory.Services.Tests.BulkUploadService.Apprenticeship
+{
+    public class LocationVariationFact
```
Comment (gunndabad) [inline Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFact.cs:5]: Pascal case property names

File: Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFacts.cs [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+                    subRegion = "",
+                    expectedError = "Validation error on row 2. Field DELIVERY_METHOD is invalid.",
+                },
+            };
+            foreach (var fact in facts)
+            {
+                Add(fact);
+            }
+
+            /* Test analysis code; uncomment this to generate a csv of all the covered variations which can then be used to check that all desired variations are covered and to facilitate discussions.
```
Comment (gunndabad) [inline Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFacts.cs:549]: If we need this here the `WriteAllText` should work cross-plat (or at least on Windows). `Path.GetTempPath` may be helpful.

File: Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFact.cs (hunk 1 above)
Comment (timabell) [reply Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFact.cs:5]: Oh yes, whoops. Fixed. Thanks!

File: Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFacts.cs (hunk 2 above)
Comment (timabell) [reply Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFacts.cs:549]: Done, thanks for the push to improve it.

**Pull request merged**